examples/camera-app/linux/src/clusters/camera-avstream-mgmt/camera-av-stream-manager.cpp

```diff
@@ -49,10 +49,85 @@ void CameraAVStreamManager::SetCameraDeviceHAL(CameraDeviceInterface * aCameraDe
 }
 
 CHIP_ERROR CameraAVStreamManager::ValidateStreamUsage(StreamUsageEnum streamUsage,
-                                                      const Optional<DataModel::Nullable<uint16_t>> & videoStreamId,
-                                                      const Optional<DataModel::Nullable<uint16_t>> & audioStreamId)
+                                                      Optional<DataModel::Nullable<uint16_t>> & videoStreamId,
+                                                      Optional<DataModel::Nullable<uint16_t>> & audioStreamId)
 {
-    // TODO: Validates the requested stream usage against the camera's resource management and stream priority policies.
+    // The server ensures that at least one stream Id has a value, and that there are streams allocated
+    // If a stream id(s) are provided, ensure that the requested Stream Usage matches the allocation
+    // If they're Null, look for a stream ID that matches the usage
+    bool matchedVideoStream = false;
+    bool matchedAudioStream = false;
+
+    // Is the requested stream usage supported by the camera?
+    auto myStreamUsages = GetCameraAVStreamMgmtServer()->GetSupportedStreamUsages();
+    auto it             = std::find(myStreamUsages.begin(), myStreamUsages.end(), streamUsage);
+    if (it == myStreamUsages.end())
+    {
+        ChipLogError(Camera, "Requested stream usage not found in supported stream usages");
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    if (videoStreamId.HasValue())
+    {
+        const std::vector<VideoStreamStruct> & allocatedVideoStreams = GetCameraAVStreamMgmtServer()->GetAllocatedVideoStreams();
+        if (videoStreamId.Value().IsNull())
+        {
+            for (const auto & stream : allocatedVideoStreams)
+            {
+                if (stream.streamUsage == streamUsage)
```

Comment:
This strict streamUsage check against the allocated stream is not needed IMO. I think, from the stream reuse perspective, we can relax this check, no? As long as the StreamUsage is supported in the StreamUsagePriorities list, it should be good.